### DISCUSSION TITLE
Raspi died and so did my "backup" ... can I restore data from a TeslaLogger backup?

### DISCUSSION THREAD

**Discussion by zwuba** [teslamate-org/teslamate#2029] [Q&A]
Dear folks,
It's been like an example out of a "backup for dummies" book but it actually happened to me that my automatic TeslaMate backup upload to my Google Drive stopped working somewhere in Feb'21 and I didn't notice ... I thought I'm safe and just kept everything as it is but then suddenly my SD-Card in my Raspi became corrupt and basically all my data from the last 7 months is gone :(
I tried to restore the local TeslaMate backup from the SD-Card but I failed  as the root.tar.xz is damaged and I just can't get the local backup data out of there ... If there's anyone capable of doing that I'm also super happy but my approach would now be the following:
I have a second Raspi running the TeslaLogger SW and there I actually have a "copy" of all of my drive/car data so I thought it might be "easier" to just import that data for the given period to the TeslaMate database if that's possible at all.
I'm happy for any input :)

**Comment by eliluong** [comment]
There is no native solution at the moment. The only supported format for importing is TeslaFi (https://docs.teslamate.org/docs/import/teslafi/). tesla-apiscraper is an old self-hosted solution, and someone wrote code to convert tesla-apiscraper's data into TeslaFi-compatible data (https://docs.teslamate.org/docs/import/tesla_apiscraper). So I think if you study the code used to convert to TeslaFi, you can write your own to convert TeslaLogger's data into a compatible format for import.
